engine/src/main/java/org/hibernate/validator/internal/util/TypeVariableBindings.java

```diff
@@ -69,7 +69,7 @@ else if ( genericSuperType instanceof ParameterizedType ) {
 
 				boolean typeParameterFoundInSubType = false;
 				for ( Entry<TypeVariable<?>, TypeVariable<?>> subTypeParameter : bindings.entrySet() ) {
-					if ( typeArgument.equals( subTypeParameter.getValue() ) ) {
+					if ( typeArgument == subTypeParameter.getValue() ) {
```

Comment:
Ok, cool. My concern is that it's relying on an implementation detail of the class library whether "==" is fine or not.